### PR TITLE
[backport] Add `mask` and `return_mask` option to `F.dropout`

### PR DESCRIPTION
**Pull request by hvy**
Backport of #4907

### CODE REVIEW

File: chainer/functions/noise/dropout.py [hunk 1]
```diff
@@ -133,11 +156,22 @@ def dropout(x, ratio=.5, **kwargs):
...
-        kwargs, train='train argument is not supported anymore. '
-        'Use chainer.using_config')
-    argument.assert_kwargs_empty(kwargs)
+    mask = None
+    return_mask = False
+    if kwargs:
+        mask, return_mask = argument.parse_kwargs(
+            kwargs, ('mask', mask), ('return_mask', return_mask),
+            train='train argument is not supported anymore. '
+                  'Use chainer.using_config')
```
Comment (hvy) [inline chainer/functions/noise/dropout.py]: For the record, manually resolved conflicts here.

Comment (kmaehashi) [reply chainer/functions/noise/dropout.py]: Could you check Jenkins errors here? Unexpected kwargs in `argument.parse_kwargs` is not backported. (#4880)

**Comment by hvy** [comment]
@kmaehashi Thanks for the quick check, I added the old check back. Does this look okay?

**Comment by kmaehashi** [comment]
Thanks, LGTM!
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**